File: Julia/Problem010.jl
# Parameters:
max = 2000000
################

primes = [2]
i = 3

while true
	for p ∈ primes
		if p^2 > i
			append!(primes, i)
			break
		elseif i % p == 0
			break
		end
	end

	global i += 2

	if i > max
		break
	end
end

println(sum(primes))

File: C++/Problem010.cpp
#include <iostream>
#include <vector>


int main() {
	// Parameters:
	long int max_ {2000000};
	///////////////

	long long int sum_ {2};

	std::vector<int> primes {};

	primes.push_back(2);
	int i {3};

	while(i < max_) {
		for(auto p : primes) {
			if(p * p > i) {
				primes.push_back(i);
				sum_ += i;
				break;
			} else if (i % p == 0) {
				break;
			}
		}

		i += 2;
	}

	std::cout << sum_ << std::endl;

	return 0;
}